src/main/resources/com/puppycrawl/tools/checkstyle/checks/coding/messages.properties

```diff
@@ -48,6 +48,7 @@ no.array.trailing.comma=Array should not contain trailing comma.
 no.enum.trailing.comma=Enum should not contain trailing comma.
 overload.methods.declaration=All overloaded methods should be placed next to each other. Previous overloaded method located at line ''{0}''.
 parameter.assignment=Assignment of parameter ''{0}'' is not allowed.
+pattern.variable.assignment=Assignment of pattern variable ''{0}'' is not allowed.
```

Comment:
Goog question. Is this assignment or reassignment?

Reply:
I think we need to change Check name. `PatternVariableReassignment`

but this searches, it is clear that "reassignment" is wide used term for this problems.

https://www.google.com/search?q=java+assignment+of+method+parameter
https://www.google.com/search?q=java+reassignment+of+method+parameter